examples/elastic/pytorch/pytorch_mnist_elastic.py

```diff
@@ -48,13 +51,14 @@
 torch.set_num_threads(1)
 
 kwargs = {'num_workers': 1, 'pin_memory': True} if args.cuda else {}
-data_dir = args.data_dir or f'data-{hvd.rank()}'
-train_dataset = \
-    datasets.MNIST(data_dir, train=True, download=True,
-                   transform=transforms.Compose([
-                       transforms.ToTensor(),
-                       transforms.Normalize((0.1307,), (0.3081,))
-                   ]))
+data_dir = args.data_dir or './data'
+with FileLock(os.path.expanduser("~/.datalock")):
```

Comment:
Can we put it relative to the datadir, e.g. as `.lock` inside the datadir?

Reply:
isn't there a download MNIST call, and that should go into the lock, not this load MNIST call?

Reply:
This won't work if the directory does not exist yet.

Reply:
Renamed to `.horovod_lock` for clarity, let me know if that makes more sense. I believe this is the only place we're downloading the MNIST data.